File: LeetCode/55JumpGame/JumpGame.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        # Greedy -> Time: O(n), Space: O(1)
        end = len(nums) - 1
        for i in range(end, -1, -1):
            if i + nums[i] >= end:
                end = i        
        return end == 0

File: LeetCode/55JumpGame/JumpGame.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        // Greedy -> Time: O(n), Space: O(1)
        int end = nums.size() - 1;
        for (int i = end; i >= 0; i--) {
            if ((i+nums[i]) >= end) {
                end = i;
            }
        }
        return (end == 0);
    }
};